### ISSUE TITLE
[error] [MJSONWP] Encountered internal error running command: TypeError: Cannot create property 'replace' on boolean 'true'

### ISSUE THREAD

**Issue by AlpBilgin** [appium/appium#10243]
## The problem

I setup Appium.app and NPM package (globally into a non sudo folder) as well as I could from official instructions. I got 
`[error] [MJSONWP] Encountered internal error running command: TypeError: Cannot create property 'replace' on boolean 'true'
`
in the first run.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.7.2
* Last Appium version that did not exhibit the issue (if applicable): -
* Desktop OS/version used to run Appium: macOS 10.13.3
* Node.js version (unless using Appium.app|exe): 
* Mobile platform/version under test: Android 8.1.0
* Real device or emulator/simulator: emulator
* Appium CLI or Appium.app|exe: Appium.app

## Details

I don't really know what could be wrong.

## Link to Appium logs

[log](https://gist.github.com/AlpBilgin/2739b96dad7fb68f39781bf3544b906d)

## Code To Reproduce Issue [ Good To Have ]

I don't really have code yet. This is the desired capabilities object that I fed to the GUI.
{
  "automationName": "UiAutomator2",
  "platformName": "Android",
  "platformVersion": "8.1.0",
  "deviceName": "Android Emulator",
  "app": "<SNIP!>",
  "avd": "Pixel_API_27"
}

**Comment by mykola-mokhnach** [comment]
Update to appium@beta

**Comment by lvconley** [comment]
I'm seeing this exact same error with Appium Desktop 1.5.0. It is blocking all automation efforts. We will now have to downgrade to 1.4.1 again to get unblocked.

**Comment by lvconley** [comment]
Here's our log:

[Appium] Welcome to Appium v1.7.2
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[HTTP] --> POST /wd/hub/session
[HTTP] {"desiredCapabilities":{"app":"/Users/laconley/Downloads/MDX_4.12_Android/DebugBuild/WDW.apk","appActivity":"com.disney.wdpro.park.activities.LoaderActivity","appPackage":"com.disney.wdw.android.debug","appiumVersion":"1.7.2","automationName":"uiautomator2","deviceName":"Nexus6P_API25","deviceOrientation":"portrait","platformName":"Android","platformVersion":"7.1.1","seleniumURL":"http://127.0.0.1:4723","newCommandTimeout":0,"connectHardwareKeyboard":true}}
[MJSONWP] Calling AppiumDriver.createSession() with args: [{"app":"/Users/laconley/Downloads/MDX_4.12_Android/DebugBuild/WDW.apk","appActivity":"com.disney.wdpro.park.activities.LoaderActivity","appPackage":"com.disney.wdw.android.debug","appiumVersion":"1.7.2","automationName":"uiautomator2","deviceName":"Nexus6P_API25","deviceOrientation":"portrait","platformName":"Android","platformVersion":"7.1.1","seleniumURL":"http://127.0.0.1:4723","newCommandTimeout":0,"connectHardwareKeyboard":true},null,null]
[BaseDriver] Event 'newSessionRequested' logged at 1522781087897 (11:44:47 GMT-0700 (PDT))
[Appium] Creating new AndroidUiautomator2Driver (v0.11.0) session
[Appium] Capabilities:
[Appium]   app: /Users/laconley/Downloads/MDX_4.12_Android/DebugBuild/WDW.apk
[Appium]   appActivity: com.disney.wdpro.park.activities.LoaderActivity
[Appium]   appPackage: com.disney.wdw.android.debug
[Appium]   appiumVersion: 1.7.2
[Appium]   automationName: uiautomator2
[Appium]   deviceName: Nexus6P_API25
[Appium]   deviceOrientation: portrait
[Appium]   platformName: Android
[Appium]   platformVersion: 7.1.1
[Appium]   seleniumURL: http://127.0.0.1:4723
[Appium]   newCommandTimeout: 0
[Appium]   connectHardwareKeyboard: true
[BaseDriver] W3C capabilities [] and MJSONWP desired capabilities  were provided
[BaseDriver] Falling back to MJSONWP desired capabilities
[BaseDriver] The following capabilities were provided, but are not recognized by appium: appiumVersion, deviceOrientation, seleniumURL, connectHardwareKeyboard.
[BaseDriver] Session created with session id: 0d255fa1-45d6-41ac-a303-ac10bd0878bd
[BaseDriver] Using local app '/Users/laconley/Downloads/MDX_4.12_Android/DebugBuild/WDW.apk'
[UiAutomator2] Checking whether app is actually present
[AndroidDriver] Getting Java version
[AndroidDriver] Java version is: 1.8.0_152
[ADB] Checking whether adb is present
[ADB] Using adb from /Users/laconley/Library/Android/sdk/platform-tools/adb
[AndroidDriver] Retrieving device list
[ADB] Trying to find a connected android device
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[AndroidDriver] Looking for a device with Android '7.1.1'
[ADB] Setting device id to emulator-5554
[ADB] Getting device platform version
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.version.release'
[ADB] Current device property 'ro.build.version.release': 7.1.1
[AndroidDriver] Using device: emulator-5554
[ADB] Checking whether adb is present
[ADB] Using adb from /Users/laconley/Library/Android/sdk/platform-tools/adb
[ADB] Setting device id to emulator-5554
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.version.sdk'
[ADB] Current device property 'ro.build.version.sdk': 25
[ADB] Device API level: 25
[ADB] Getting device platform version
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.version.release'
[ADB] Current device property 'ro.build.version.release': 7.1.1
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell wm size'
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell wm density'
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.product.model'
[ADB] Current device property 'ro.product.model': Android SDK built for x86
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.product.manufacturer'
[ADB] Current device property 'ro.product.manufacturer': Google
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell am force-stop io.appium.uiautomator2.server'
[ADB] Checking whether aapt is present
[ADB] Using aapt from /Users/laconley/Library/Android/sdk/build-tools/27.0.2/aapt
[ADB] Getting install status for io.appium.uiautomator2.server
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pm list packages io.appium.uiautomator2.server'
[ADB] App is not installed
[ADB] Getting install status for io.appium.uiautomator2.server.test
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pm list packages io.appium.uiautomator2.server.test'
[ADB] App is not installed
[ADB] Checking app cert for /Applications/Appium.app/Contents/Resources/app/node_modules/appium-uiautomator2-driver/uiautomator2/appium-uiautomator2-server-v0.3.0.apk
[ADB] Using apksigner from /Users/laconley/Library/Android/sdk/build-tools/27.0.2/apksigner
[ADB] Using apksigner from /Users/laconley/Library/Android/sdk/build-tools/27.0.2/apksigner
[ADB] Starting '/Users/laconley/Library/Android/sdk/build-tools/27.0.2/apksigner' with args 'verify,--print-certs,/Applications/Appium.app/Contents/Resources/app/node_modules/appium-uiautomator2-driver/uiautomator2/appium-uiautomator2-server-v0.3.0.apk'
[ADB] '/Applications/Appium.app/Contents/Resources/app/node_modules/appium-uiautomator2-driver/uiautomator2/appium-uiautomator2-server-v0.3.0.apk' is already signed.
[UiAutomator2] Deleting UiAutomator2 session
[UiAutomator2] Deleting UiAutomator2 server session
[UiAutomator2] Did not get confirmation UiAutomator2 deleteSession worked; Error was: Error: Trying to proxy a session command without session id
[ADB] Getting connected devices...
[ADB] 1 device(s) connected
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell am force-stop com.disney.wdw.android.debug'
[ADB] Removing forwarded port socket connection: 8200 
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 forward --remove tcp\:8200'
[ADB] Running '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 forward --remove tcp\:8200'
[UiAutomator2] Unable to remove port forward 'Error executing adbExec. Original error: 'Command '/Users/laconley/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 forward --remove tcp\:8200' exited with code 1'; Stderr: 'error: listener 'tcp:8200' not found'; Code: '1''
[MJSONWP] Encountered internal error running command: TypeError: Cannot create property 'replace' on boolean 'true'
[MJSONWP]     at ADB.callee$0$0$ (../../../lib/tools/apk-utils.js:336:20)
[MJSONWP]     at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:67:40)
[MJSONWP]     at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:315:22)
[MJSONWP]     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:100:21)
[MJSONWP]     at invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:136:37)
[MJSONWP]     at enqueueResult (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:185:17)
[MJSONWP]     at Promise ()
[MJSONWP]     at F (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/core-js/library/modules/$.export.js:30:36)
[MJSONWP]     at AsyncIterator.enqueue (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:184:12)
[MJSONWP]     at AsyncIterator.prototype.(anonymous function) [as next] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:100:21)
[MJSONWP]     at Object.runtime.async (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-adb/node_modules/babel-runtime/regenerator/runtime.js:209:12)
[MJSONWP]     at ADB.callee$0$0 [as install] (../../../lib/tools/apk-utils.js:361:23)
[MJSONWP]     at UiAutomator2Server.signAndInstall$ (../../lib/uiautomator2.js:109:20)
[MJSONWP]     at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-uiautomator2-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
[MJSONWP]     at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-uiautomator2-driver/node_modules/babel-runtime/regenerator/runtime.js:315:22)
[MJSONWP]     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-uiautomator2-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
[MJSONWP]     at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium-uiautomator2-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
[MJSONWP]     at 
[HTTP] <-- POST /wd/hub/session 500 2315 ms - 187
[HTTP] 
Appium server stopped successfully

**Comment by mykola-mokhnach** [comment]
@dpgraham FYI

**Comment by dpgraham** [comment]
@lvconley Can you try running the Appium CLI instead of from the Desktop UI and see if that works?

**Comment by lvconley** [comment]
@dpgraham Hi, we don't have any issues running our code against 1.7.2 headless server at all, on either Android or iOS simulators. We got a new build for automating today that had a whole new gui--completely different from our previous release, so we needed Appium Desktop to start creating the new page objects, etc, but found that no one on the team could launch the new app on an Android 7.1.1 AVD via the 1.5.0 version of Appium Desktop. So, after much googling, bing-ing, and tweaking of capabilities, the whole team downgraded to Desktop 1.4.1 and we are now unblocked.

**Comment by lvconley** [comment]
You know, I just now noticed something funny and possibly useful-- a folder called "node_modules" has been created in the root of my Java/mvn project! It should not be there (its existence actually blew a branch verification earlier in the day that prevented a merge, but I didn't have time to investigate at the time), but it had to have been created there due to Appium Desktop, somehow!

**Comment by lvconley** [comment]
OK, here's a screenshot of my Eclipse Project Explorer showing the directory:

<img width="230" alt="projectcontamination" src="https://user-images.githubusercontent.com/13697495/38322486-6958f4ec-37ef-11e8-91fd-94f6c9a93b79.png">

I think this appeared after I tried running code against Appium 1.5.0 Desktop's built-in 1.7.2 server. The other colleague who experienced the same errors I was having, actually only tried to launch our app via the Desktop Inspector and does not have this node_modules folder in her project root as I do.

**Comment by dpgraham** [comment]
@lvconley None of these files look familiar to me, I'm not sure they're Appium related. Can you have a look inside the README.md and let me know what you see?

Also, I would recommend that you just add `node_modules` to your .gitignore (if you're using Git, I don't know the equivalents in other source control). `node_modules` contains dependencies that are built by Node, and don't need to be checked in.

**Comment by lvconley** [comment]
Here's what's in the Read Me:

<img width="743" alt="screen shot 2018-04-04 at 10 38 50 am" src="https://user-images.githubusercontent.com/13697495/38324210-74099efa-37f4-11e8-99ce-ce66c212a09b.png">

I did use NPM to uninstall Appium and then re-install, as I explored whether or not there was a newer version of the headless server available. NPM must then have placed the folder there? This folder was placed where it is programmatically, not manually.
